src/fields/composed_quadrature.cc

```diff
@@ -0,0 +1,72 @@
+/*!
+ *
+﻿ * Copyright (C) 2015 Technical University of Liberec.  All rights reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License version 3 as published by the
+ * Free Software Foundation. (http://www.gnu.org/licenses/gpl-3.0.en.html)
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+ *
+ *
+ * @file    composed_quadrature.cc
+ * @brief
+ * @author  David Flanderka
+ */
+
+#include "fields/composed_quadrature.hh"
+#include "fields/point_sets.hh"
+#include "quadrature/quadrature.hh"
+#include <memory>
+
+
+template <unsigned int dim>
+EvalPoints<dim>::EvalPoints()
+{}
+
+template <unsigned int dim>
+BulkSubQuad<dim> EvalPoints<dim>::add_bulk(const Quadrature<dim> &quad)
+{
+    ASSERT(bulk_set_.c_quad_==nullptr).error("Multiple initialization of bulk point set!\n");
+
+    bulk_set_.c_quad_ = this;
+    for (auto p : quad.get_points()) bulk_set_.point_indices_.push_back( this->add_local_point(p) );
```

Comment:
Before push_back() we should first check for duplicity and add only if the point is not in the point set. The same below in add_side().